packages/graph/src/graph2/graph.ts

```diff
@@ -209,6 +200,22 @@ export class Graph2 extends SVGZoomWidget {
         this.zoomToFitLimit(1);
     }
 
+    resolveAnnoEventNode(): undefined | IVertex4Annotation {
+        const d3evt = d3Event();
+        const eventPath = d3evt?.sourceEvent?.path ?? d3evt.path;
+        const anno = eventPath.find(n => {
+            if(typeof n.hasAttribute === "function") {
+                return !!n.hasAttribute("data-anno");
+            }
+            return false;
+        });
+        let annoData;
+        if(anno){
+            annoData = JSON.parse(anno.getAttribute("data-anno"));
+        }
+        return annoData;
```

Comment:
Minor:  Could could be reduced to  `return anno ? JSON.parse(anno.getAttribute("data-anno")) : undefined`